pkg/domainutil/domain.go

```diff
@@ -0,0 +1,59 @@
+package domainutil
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+const (
+	DefaultDomain = "astronomer.io"
+	LocalDomain   = "localhost"
+)
+
+var (
+	// CloudDomainRegex is used to differentiate cloud domain from software domain
+	// See https://github.com/astronomer/astrohub-cli/issues/7 for regexp rationale
+	// This will need to be handled as part of the permanent solution to issue #432
+	CloudDomainRegex     = regexp.MustCompile(`(cloud\.|^)astronomer(?:(-dev|-stage|-perf))?\.io`)
+	PRPreviewDomainRegex = regexp.MustCompile(`(pr\d{4}\.)astronomer(-dev)\.io$`)
+)
+
+func FormatDomain(domain string) string {
+	if strings.Contains(domain, "cloud") {
+		domain = strings.Replace(domain, "cloud.", "", 1)
+	} else if domain == "" {
+		domain = DefaultDomain
+	}
+
+	return domain
+}
+
+func isPrPreviewDomain(domain string) bool {
+	return PRPreviewDomainRegex.MatchString(domain)
+}
+
+func getPRSubDomain(domain string) (prSubDomain, restOfDomain string) {
+	if isPrPreviewDomain(domain) {
+		prSubDomain, domain, _ = strings.Cut(domain, ".")
```

Comment:
yeah so this could be replaced with the `re.FindStringSubmatch`

Reply:
true but we still need to mutate the domain and extract the pr-number part and the domain part as we use them differently in `GetURLToEndpoint()`